src/pins/pin.rs

```diff
@@ -175,6 +175,20 @@ pub struct Pin<T: Trait, S: State> {
     pub(crate) _state: S,
 }
 
+/// A drop-in substitute for the `PIO0_0`, `PIO0_1`... etc. structs which are generated at compile
+/// time and used to bind each pin to a [`pins::Token`].
+/// `GenericPin`s are *only* intended for use cases which are made impossible by the default
+/// [`pins::Trait`] having pin information compiled into its type, e.g. if you'd like to store
+/// several pin instances in a collection, or modify them in batches.
+///
+/// [`pins::Token`]: struct.Token.html
+/// [`pins::Trait`]: trait.Trait.html
+#[derive(Clone, Copy)]
```

Comment:
This must not be `Clone` or `Copy`, or the uniqueness guarantee no longer holds!

```suggestion
```